src/js/common/parser.js

```diff
@@ -0,0 +1,223 @@
+/*
```

Comment:
This large comment needs to go into some proper docs. I am thinking that we need a more structured summary of the errors rather than simply reporting the human-readable message. The user will need some means of determining, somehow, WHICH validation rule, in terms of the schema, was responsible for the message.